pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -827,39 +826,41 @@ public CompletableFuture<Void> connectionOpened(final ClientCnx cnx) {
         final CompletableFuture<Void> future = new CompletableFuture<>();
         synchronized (this) {
             setClientCnx(cnx);
-            ByteBuf request = Commands.newSubscribe(topic, subscription, consumerId, requestId, getSubType(),
-                    priorityLevel, consumerName, isDurable, startMessageIdData, metadata, readCompacted,
-                    conf.isReplicateSubscriptionState(),
-                    InitialPosition.valueOf(subscriptionInitialPosition.getValue()),
-                    startMessageRollbackDuration, si, createTopicIfDoesNotExist, conf.getKeySharedPolicy(),
-                    // Use the current epoch to subscribe.
-                    conf.getSubscriptionProperties(), CONSUMER_EPOCH.get(this));
-
-            cnx.sendRequestWithId(request, requestId).thenRun(() -> {
-                synchronized (ConsumerImpl.this) {
-                    if (changeToReadyState()) {
-                        consumerIsReconnectedToBroker(cnx, currentSize);
-                    } else {
-                        // Consumer was closed while reconnecting, close the connection to make sure the broker
-                        // drops the consumer on its side
-                        setState(State.Closed);
-                        deregisterFromClientCnx();
-                        client.cleanupConsumer(this);
-                        cnx.channel().close();
-                        future.complete(null);
-                        return;
+            final SchemaInfo finalSi = si;
+            sendCloseConsumerCommand(cnx).thenCompose(ignore -> {
```

Comment:
This is a fundamental change to the protocol. It feels like there must be a bug somewhere else.